app/modules/report/audio_event_report.rb

```diff
@@ -0,0 +1,552 @@
+      ReportQuery.new(table, cte)
+    end
+
+    def composition_series_aggregated_for_main_query(composition_series)
+      composition_series_aliased = composition_series.table.as('c')
+      composition_series_json = Arel::SelectManager.new
+        .project(composition_series_aliased.right.json_agg)
+        .from(composition_series_aliased)
+    end
+
+    # @param filter_params [ActionController::Parameters] the filter parameters
+    # @param base_scope [ActiveRecord::Relation] the base scope for the query
+    def filter_as_relation(filter_params, base_scope)
+      filter_query = Filter::Query.new(
+        filter_params,
+        base_scope,
+        AudioEvent,
+        AudioEvent.filter_settings
+      )
+      filter_query.query_without_paging_sorting
+    end
+
+    def audio_events = AudioEvent.arel_table
+    def audio_recordings = AudioRecording.arel_table
+    def sites = Site.arel_table
+    def regions = Region.arel_table
+    def tags = Tag.arel_table
+    def taggings = Tagging.arel_table
+    def provenance = Provenance.arel_table
+    def verifications = Verification.arel_table
+
+    # Default attributes for projection
+    # @return [Array<Arel::Attributes>] the attributes to select
+    def attributes
+      [
+        taggings[:id].as('tagging_ids'),
+        sites[:id].as('site_ids'),
+        regions[:id].as('region_ids'),
+        tags[:id].as('tag_id'),
+        audio_events[:audio_recording_id].as('audio_recording_ids'),
+        audio_events[:provenance_id].as('provenance_ids'),
+        audio_events[:score].as('score'),
+        audio_events[:id].as('audio_event_id'),
+        audio_recordings[:recorded_date],
+        audio_recordings[:audio_recording_id].as('audio_recording_id'),
+        audio_recordings[:duration_seconds],
+        # verifications[:id].as('verification_id'),
+        # audio event absolute start and end time
+        Arel::Nodes::SqlLiteral.new(start_time_absolute_expression),
+        Arel::Nodes::SqlLiteral.new(end_time_absolute_expression)
+      ]
+    end
+
+    # Adds left joins to the query
+    def add_joins(query)
+      query
+        .join(taggings)
+        .on(audio_events[:id].eq(taggings[:audio_event_id]))
+        .join(tags)
+        .on(taggings[:tag_id].eq(tags[:id]))
+        .join(regions, Arel::Nodes::OuterJoin)
+        .on(regions[:id].eq(sites[:region_id]))
+    end
+
+    # returns an expression that projects the absolute start time of an audio
+    # event as a derived columnn
+    def start_time_absolute_expression
+      'audio_recordings.recorded_date + CAST(audio_events.start_time_seconds || \' seconds\' as interval) ' \
+        'as start_time_absolute'
+    end
+
+    # returns an expression that projects the absolute end time of an audio
+    # event as a derived columnn
+    def end_time_absolute_expression
+      'audio_recordings.recorded_date + CAST(audio_events.end_time_seconds || \' seconds\' as interval) ' \
+        'as end_time_absolute'
+    end
```

Comment:
pretty sure these already exist elsewhere in the solution

helpers like this can go on the AudioEvent model